File: android-maps/src/main/java/com/lightningkite/kwift/maps/android/LatLng.ext.kt
package com.lightningkite.kwift.maps.android

import com.google.android.gms.maps.model.LatLng
import com.lightningkite.kwift.location.shared.GeoCoordinate

fun GeoCoordinate.toMaps(): LatLng = LatLng(latitude, longitude)
fun LatLng.toKwift(): GeoCoordinate =
    GeoCoordinate(latitude, longitude)

File: web-maps-leaflet/src/LatLng.ext.ts

import {GeoCoordinate} from "khrysalis/dist/location/GeoCoordinate.shared";
import {LatLng} from "leaflet";

//! Declares com.lightningkite.khrysalis.maps.toMaps
export function xLatLngToMaps(this_: LatLng): GeoCoordinate {
    return new GeoCoordinate(this_.lat, this_.lng);
}
//! Declares com.lightningkite.khrysalis.maps.toKhrysalis
export function xGeoCoordinateToKhrysalis(this_: GeoCoordinate): LatLng {
    return new LatLng(this_.latitude, this_.longitude);
}
